src/raster/mdarray.rs

```diff
@@ -0,0 +1,331 @@
+    fn test_dimension_size() {
+        let dataset_options = DatasetOptions {
+            open_flags: GdalOpenFlags::GDAL_OF_MULTIDIM_RASTER,
+            allowed_drivers: None,
+            open_options: None,
+            sibling_files: None,
+        };
+        let dataset = Dataset::open_ex("fixtures/byte_no_cf.nc", dataset_options).unwrap();
+        let root_group = dataset.root_group().unwrap();
+        let array_name = "Band1".to_string();
+        let options = cpl::CslStringList::new(); //Driver specific options determining how the array should be opened. Pass nullptr for default behavior.
+        let md_array = root_group.md_array(array_name, options);
+        let dimensions = md_array.get_dimensions().unwrap();
+        let mut dimensions_size = Vec::new();
+        for dimension in dimensions {
+            dimensions_size.push(dimension.size());
+        }
+        assert_eq!(dimensions_size, vec![20, 20])
+    }
+}
+
+/// Represent an MDArray in a Group
+///
+/// This object carries the lifetime of the Group that
+/// contains it. This is necessary to prevent the Group
+/// from being dropped before the mdarray.
+#[derive(Debug)]
+pub struct MDArray<'a> {
+    c_mdarray: *mut GDALMDArrayHS, //H
+    group: &'a Group<'a>,
+}
+
+#[allow(dead_code)]
+pub struct Dimension<'a> {
+    c_dimension: *mut GDALDimensionHS,
+    md_array: &'a MDArray<'a>,
+}
+
+impl<'a> Dimension<'a> {
+    /// Create a MDArray from a wrapped C pointer
+    ///
+    /// # Safety
+    /// This method operates on a raw C pointer
+    pub fn from_c_dimension(md_array: &'a MDArray<'a>, c_dimension: *mut GDALDimensionHS) -> Self {
+        Dimension {
+            c_dimension: (c_dimension),
+            md_array: (md_array),
+        }
+    }
+    pub fn size(self) -> usize {
+        unsafe { GDALDimensionGetSize(self.c_dimension) as usize }
+    }
+
+    pub fn name(self) -> String {
+        _string(unsafe { GDALDimensionGetName(self.c_dimension) })
+    }
+}
+
+impl<'a> MDArray<'a> {
+    /// Create a MDArray from a wrapped C pointer
+    ///
+    /// # Safety
+    /// This method operates on a raw C pointer
+    pub fn from_c_mdarray(group: &'a Group, c_mdarray: *mut GDALMDArrayHS) -> Self {
+        MDArray { c_mdarray, group }
+    }
+
+    pub fn n_dimension(&self) -> usize {
+        unsafe { GDALMDArrayGetDimensionCount(self.c_mdarray) }
+    }
+
+    pub fn n_elements(&self) -> u64 {
+        unsafe { GDALMDArrayGetTotalElementsCount(self.c_mdarray) }
+    }
+
+    pub fn get_dimensions(&self) -> Result<Vec<Dimension>> {
+        // break on ndims not is_null
+        let n_dimension = self.n_dimension();
+        unsafe {
+            let mut pn_count: usize = 0;
+            let pn_count_ptr: *mut usize = &mut pn_count;
+            let c_dimensions = GDALMDArrayGetDimensions(self.c_mdarray, pn_count_ptr);
+            // if c_group.is_null() {
+            //     return Err(_last_null_pointer_err("GDALGetRasterBand"));
+            // }
```

Comment:
If there is no possibility of a null pointer, you can remove it. Otherwise, we should integrate it.